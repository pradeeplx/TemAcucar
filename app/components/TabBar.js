import React, { Component } from 'react'
import { View, Text, TouchableOpacity, Dimensions, StyleSheet } from 'react-native'
import Colors from "../Colors"
import Sentence from "./Sentence"
import Icon from "./Icon"

export default class TabBar extends Component {
  constructor(props, context) {
    super(props, context)
    this.selectedTabIcons = []
    this.unselectedTabIcons = []
  }

  renderTabOption(name, page) {
    const { notificationsCount } = this.props
    const isActive = (this.props.activeTab === page)
    const height = Dimensions.get('window').height
    const width = Dimensions.get('window').width
    return (
      <TouchableOpacity key={name} onPress={() => this.props.goToPage(page)} style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        height: 60,
        backgroundColor: (isActive ? Colors.white : Colors.white),
      }}>
        <Icon name={name} size={36} color={isActive ? Colors.pink : Colors.gray} />
        { page === 2 && notificationsCount > 0 &&
          <View style={{
            position: 'absolute',
            right: width * 0.11,
            top: height * 0.015,
            height: 18,
            width: 18,
            backgroundColor: (isActive ? Colors.white : Colors.pink),
            borderWidth: StyleSheet.hairlineWidth,
            borderColor: (isActive ? Colors.pink : Colors.pink),
            borderRadius: 9,
          }}>
            <Sentence style={{
              backgroundColor: 'transparent',
              color: (isActive ? Colors.pink : Colors.white),
              textAlign: 'center',
              fontSize: 12,
            }}>
              {notificationsCount}
            </Sentence>
          </View>
        }
      </TouchableOpacity>
    );
  }

  render() {
    return (
      <View elevation={3} style={{
        backgroundColor: Colors.white,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        shadowColor: 'black',
        shadowOpacity: 0.3,
        shadowRadius: 2,
        shadowOffset: {
          height: 1,
          width: 0,
        },
        overflow: 'visible',
        transform: [{'translate': [0,0,1]}],
        borderColor: Colors.gray,
        borderBottomWidth: StyleSheet.hairlineWidth,
      }}>
        {this.props.tabs.map((tab, i) => this.renderTabOption(tab, i))}
      </View>
    )
  }
}

TabBar.propTypes = {
  goToPage: React.PropTypes.func,
  activeTab: React.PropTypes.number,
  tabs: React.PropTypes.array,
}
