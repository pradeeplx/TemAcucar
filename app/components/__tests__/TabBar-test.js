import 'react-native'
import React from 'react'
import TabBar from '../TabBar'
import renderer from 'react-test-renderer'

const props = {
  notificationsCount: 0,
  activeTab: 1,
  tabs: ['ios-barcode', 'ios-battery-charging', 'ios-cafe'],
}

it('renders correctly', () => {
  const tree = renderer.create(
    <TabBar {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders notificationsCount', () => {
  const tree = renderer.create(
    <TabBar {...props} notificationsCount={2} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
