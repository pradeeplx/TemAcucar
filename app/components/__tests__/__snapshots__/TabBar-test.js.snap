exports[`test renders correctly 1`] = `
<View
  elevation={3}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 0.5,
      "borderColor": "#CECED2",
      "flexDirection": "row",
      "justifyContent": "center",
      "overflow": "visible",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "transform": Array [
        Object {
          "translate": Array [
            0,
            0,
            1,
          ],
        },
      ],
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#CECED2",
            "fontFamily": "Ionicons",
            "fontSize": 36,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FF7F85",
            "fontFamily": "Ionicons",
            "fontSize": 36,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#CECED2",
            "fontFamily": "Ionicons",
            "fontSize": 36,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
  </View>
</View>
`;

exports[`test renders notificationsCount 1`] = `
<View
  elevation={3}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 0.5,
      "borderColor": "#CECED2",
      "flexDirection": "row",
      "justifyContent": "center",
      "overflow": "visible",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "transform": Array [
        Object {
          "translate": Array [
            0,
            0,
            1,
          ],
        },
      ],
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#CECED2",
            "fontFamily": "Ionicons",
            "fontSize": 36,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FF7F85",
            "fontFamily": "Ionicons",
            "fontSize": 36,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#CECED2",
            "fontFamily": "Ionicons",
            "fontSize": 36,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#FF7F85",
          "borderColor": "#FF7F85",
          "borderRadius": 9,
          "borderWidth": 0.5,
          "height": 18,
          "position": "absolute",
          "right": 82.5,
          "top": 20.009999999999998,
          "width": 18,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontSize": 12,
              "textAlign": "center",
            },
          ]
        }>
        2
      </Text>
    </View>
  </View>
</View>
`;
